src/boot/bootctl.c

```diff
@@ -1647,6 +1653,18 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_arch_all = true;
                         break;
 
+                case ARG_EFI_LABEL:
+                        r = free_and_strdup(&arg_efi_label, optarg);
+                        if (r < 0)
+                                return log_oom();
+                        if (isempty(arg_efi_label) || !string_is_safe(arg_efi_label))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "Invalid --efi-label=: %s", arg_efi_label);
+                        if (strlen(arg_efi_label) > 1024)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "--efi-label= too long: %ld > 1024", strlen(arg_efi_label));
```

Comment:
Please first check the string by `optarg`, then copy it.

Please use `%zu` instead of `%ld` for `strlen()`.

I do not know the validity of the limit 1024. Where does it come from?

Reply:
> I do not know the validity of the limit 1024. Where does it come from?

It was choosen arbitrarily by me. Having another thought I am not sure the code here should be in the business of length checking given the UEFI spec doesn't specify any upper limit (see resolved comment in previous round of review). So I removed it again...
I am assuming when communicating with the firmware it will fail gracefully and return a proper error in case the string is too long.

Reply:
I see. But 1024 is already too long, I guess. And even we do not know the exact limit, we should set a limit to not break firmware.

@medhefgo and @poettering WDYT?

Reply:
Afaik the kernel will enforce a limit to the var first, but it makes sense to limit this to something sensible anyway.

Reply:
> limit this to something sensible anyway.

Lowered the limit to 250. Is this more sensible? (I don't even have a reference for how large NVRAM is on average and on average how much of it is used).

Reply:
For arbitrarily chosen numeric limits we prefer if macros are added. i.e. add at some appropriate place:

```c
#define EFI_BOOT_OPTION_DESCRIPTION_MAX 250
```

And maybe add a comment saying where the 250 comes from. Maybe something "UEFI Spec doesn't put a limit on this, but let's use something resonably conservative, that would fit into a Linux file name".

any reason to pick 250 instead of 255 btw?

Reply:
> any reason to pick 250 instead of 255 btw?

no. i put 255 again (i think i had it earlier, but when changing the numbers around last i put was 250)